status-handler.js

```diff
@@ -54,6 +56,32 @@ class StatusHandler {
   async event(data) {
     const { StatusEvent, StatusHead, Status } = this.models;
     const ev = this._transform(data, 'event');
+
+    // Fetch list of previous events for a certain build
```

Comment:
pull this out to a separate function and only run the logic if webhooks are enabled.